ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -755,6 +758,23 @@ public static long getTimeInMsToTheNearestSec(long timeInMs) {
   }
 
   /**
+   * Delete a directory recursively.
+   * @param file the directory to delete.
+   * @return {@code true} if successful, {@code false} if all files could not be deleted.
+   */
+  public static boolean deleteDirectory(File file) {
+    File[] contents = file.listFiles();
```

Comment:
It is null for normal files, but I changed this implementation to something safer (that doesn't follow symlinks).